Clean_up related to symex_assign_symbol

romainbrenguier opened the pull request:
There is no functional changes, this is just aimed at making the code a bit simpler

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [na] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [na] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [na] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

romainbrenguier committed:
Inline variable used only once

This intermediary variable doesn't help understanding and slows down
reading the code.

romainbrenguier committed:
Simplify assignment of l2_rhs

romainbrenguier committed:
Declare variable closer to where it is used

Makes it clearer why it is declared.

romainbrenguier committed:
Inline variable that is immediately moved

romainbrenguier committed:
Remove misleading comment

ns.lookup doesn't do the assignment

romainbrenguier committed:
Inline reference used only once

The definition of this reference is not useful.

romainbrenguier committed:
Infer hidden field in SSA_assignment_stept

We can deduce it from the assignment type rather than having to give it
to the constructor.

romainbrenguier committed:
Infer cond_expr in SSA_assignment_step

This can be deduced fro lhs and rhs without having to give it to the
constructor.

romainbrenguier committed:
Inline variables used immediatly after

Declaring variables for these values doesn't add clarity and makes the
function longer.

romainbrenguier committed:
Clean-up increase_generation

Add missing const, necessary narrow and insert if and else to make r_opt
scoped to only the if branch.

romainbrenguier committed:
Add missing slash

No functional changes.
Necessary for doxygen comment

romainbrenguier committed:
Document l2_thread_write_encoding

romainbrenguier committed:
Formatting changes suggested by clang-format

This is necessary for CI

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: bc044d93300232080331ce144a807ef78a7981bd).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/115563600

tautschnig commented on src/goto-symex/symex_assign.cpp:
```diff
@@ -434,16 +434,10 @@ void goto_symext::symex_assign_symbol(
-    }
-    return state.rename(std::move(guarded_rhs), ns).get();
-  }();
+    guard.empty() ? rhs : (exprt)if_exprt{conjunction(guard), rhs, lhs}, ns)
```
I'd prefer `static_cast<exprt>` as the C-style cast is just very easy to overlook.

tautschnig commented on src/goto-symex/ssa_step.cpp:
```diff
@@ -231,7 +230,7 @@ SSA_assignment_stept::SSA_assignment_stept(
   ssa_rhs = std::move(_ssa_rhs);
   assignment_type = _assignment_type;
-  cond_expr = std::move(_cond_expr);
+  cond_expr = equal_exprt(ssa_lhs, ssa_rhs);
```
Nit pick: use curly braces

tautschnig commented on src/goto-symex/symex_assign.cpp:
```diff
@@ -475,15 +475,11 @@ void goto_symext::symex_assign_symbol(
-    make_and(state.guard.as_expr(), conjunction(guard));
-
-  const exprt original_lhs = get_original_name(l2_full_lhs);
   target.assignment(
```
Random place: typo in the commit message: "this values" -> "these values", and also it should be "Inline variables" (plural)

tautschnig commented on src/goto-symex/goto_state.cpp:
```diff
@@ -31,19 +31,17 @@ std::size_t goto_statet::increase_generation(
   {
-    level2.current_names.insert(l1_identifier, std::make_pair(lhs, n));
+    std::pair<ssa_exprt, unsigned> copy = r_opt->get();
+    copy.second = narrow<unsigned>(n);
```
Ack, but we should really stop using `unsigned`, the renaming index ought to be `std::size_t`.

romainbrenguier replied:
@tautschnig I added that change in a new commit

tautschnig replied:
Thank you!

tautschnig commented on src/goto-symex/goto_symex_state.h:
```diff
@@ -217,7 +217,7 @@ class goto_symex_statet final : public goto_statet
 
   /// Allocates a fresh L2 name for the given L1 identifier, and makes it the
-  //  latest generation on this path.
+  /// latest generation on this path.
```
Note on the commit message: that's a "slash", not a "backslash"

romainbrenguier committed:
Make index in renamings be of std::size_t type

This avoids some conversions, in particular when assigning from the
return value of fresh_l2_name_provider.

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 71caf32f40eb7701d1072807d19ca19991d86955).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/115789758
